dmc/deployments/services/heroku_service/api_heroku_service.rb

```diff
@@ -1,7 +1,49 @@
 module Deployments
   class ApiHerokuService
+    include CommandRunner
 
-    def deploy(deployment)
+    def deploy(project, environment, deployment)
+      run(project, environment, deployment) do |runner|
+        runner.execute "Check access on Heroku app: #{environment.heroku_app_name}", &check_access
+        runner.execute "Deploy on #{environment.heroku_app_name}", &push
+      end
+    end
+
+    private
+
+    def check_access
```

Comment:
Why not define `check_access(log)` and use `runner.execute "Check access on Heroku app: #{environment.heroku_app_name}", &method(:check_access)`?